panic: runtime error: makeslice: len out of range

Mon Mar 30 22:03:47:Downloads rstackhouse$ exercism fetch cpp

panic: runtime error: makeslice: len out of range

goroutine 1 [running]:
strings.Repeat(0x38c6b0, 0x1, 0xfffffffffffffffb, 0x0, 0x0)
	/usr/local/go/src/strings/strings.go:458 +0x68
github.com/exercism/cli/user.(*Homework).heading(0xc20800ade0, 0x3, 0x1, 0x9)
	/Users/kytrinyx/code/go/src/github.com/exercism/cli/user/homework.go:115 +0x25e
github.com/exercism/cli/user.(*Homework).Report(0xc20800ade0, 0x3)
	/Users/kytrinyx/code/go/src/github.com/exercism/cli/user/homework.go:84 +0xbe
github.com/exercism/cli/user.(*Homework).Summarize(0xc20800ade0, 0x0)
	/Users/kytrinyx/code/go/src/github.com/exercism/cli/user/homework.go:137 +0x64
github.com/exercism/cli/cmd.Fetch(0xc20805a160)
	/Users/kytrinyx/code/go/src/github.com/exercism/cli/cmd/fetch.go:39 +0x709
github.com/codegangsta/cli.Command.Run(0x397270, 0x5, 0x3970f0, 0x1, 0x43cb30, 0x65, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/kytrinyx/code/go/src/github.com/codegangsta/cli/command.go:101 +0xe42
github.com/codegangsta/cli.(*App).Run(0xc208070000, 0xc20800a000, 0x3, 0x3, 0x0, 0x0)
	/Users/kytrinyx/code/go/src/github.com/codegangsta/cli/app.go:130 +0xb81
main.main()
	/Users/kytrinyx/code/go/src/github.com/exercism/cli/exercism/main.go:149 +0xb3f

goroutine 8 [IO wait]:
net.(*pollDesc).Wait(0xc208010290, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208010290, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc208010230, 0xc2080a0000, 0x1000, 0x1000, 0x0, 0x711b88, 0xc20802b100)
	/usr/local/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc20802e060, 0xc2080a0000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:121 +0xdc
net/http.noteEOFReader.Read(0x7131f0, 0xc20802e060, 0xc20805a318, 0xc2080a0000, 0x1000, 0x1000, 0x2e8460, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1270 +0x6e
net/http.(*noteEOFReader).Read(0xc20801e760, 0xc2080a0000, 0x1000, 0x1000, 0xc208012000, 0x0, 0x0)
	<autogenerated>:125 +0xd4
bufio.(*Reader).fill(0xc20805c660)
	/usr/local/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).Peek(0xc20805c660, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xf0
net/http.(*persistConn).readLoop(0xc20805a2c0)
	/usr/local/go/src/net/http/transport.go:842 +0xa4
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0xc9f

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 9 [select]:
net/http.(*persistConn).writeLoop(0xc20805a2c0)
	/usr/local/go/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0xcbc

goroutine 12 [runnable]:
net/http.(*persistConn).readLoop(0xc20805a580)
	/usr/local/go/src/net/http/transport.go:928 +0x9ce
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0xc9f

goroutine 13 [select]:
net/http.(*persistConn).writeLoop(0xc20805a580)
	/usr/local/go/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0xcbc

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,7 @@ The exercism CLI follows [semantic versioning](http://semver.org/).
 
 ## Next Release
 
+* [#174](https://github.com/exercism/cli/issues/174): Fix panic during fetch
 * **Your contribution here**
 
 ## v2.0.1 (2015-25-05)

--- a/user/homework.go
+++ b/user/homework.go
@@ -112,7 +112,10 @@ func (hw *Homework) heading(filter HWFilter, count, width int) {
 		status = "Not Submitted:"
 	}
 	summary := fmt.Sprintf("%d %s", count, unit)
-	padding := strings.Repeat(" ", width-len(status))
+	var padding string
+	if width > len(status) {
+		padding = strings.Repeat(" ", width-len(status))
+	}
 	fmt.Printf(hw.template, status, padding, summary)
 }
 

